universe/Condition.cpp

```diff
@@ -5110,12 +5111,22 @@ namespace {
             if (!candidate)
                 return false;
 
-            // is it a ship?
-            auto ship = std::dynamic_pointer_cast<const ::Ship>(candidate);
-            if (!ship)
-                return false;
+            auto fighter = std::dynamic_pointer_cast<const ::Fighter>(candidate);
+            const ShipDesign* design = nullptr;
+            if (fighter) {
+                // it is a fighter
+                auto carriers = Objects().FindObjects<Ship>( (std::vector<int> ) { fighter->LaunchedFrom() } );
```

Comment:
Just use `Objects().Object<Ship>(fighter->LaunchedFrom())` to get a single ship.